ReactNativeClient/lib/reducer.js

```diff
@@ -755,7 +755,8 @@ const reducer = (state = defaultState, action) => {
 		case 'SEARCH_ADD':
 			{
 				newState = Object.assign({}, state);
-				const searches = newState.searches.slice();
+				let searches = newState.searches.slice();
+        searches = searches.filter(s => s.title !== action.search.title);
```

Comment:
Why not just `if (!searches.includes(action.search)) searches.push(action.search)`? That way you don't need to remove then add the element.

Reply:
I also thought about that. But the issue with that approach is that we lose the ordering by recency. For example, we search for `apple`, `banana`, `orange` and then we search for `apple` again. In the first approach nothing will change and we can't know which one is the most recent search: `[apple, banana, orange]`. In my approach, we preserve the sorting by recency, so that the more recent the search,  the lower it will be in the list: `[banana, orange, apple]`. In conclusion, `filter` allows us to see the exact history of elements.

Reply:
Thanks for clarifying, your explanation makes sense and we should indeed add the new entry to the top.

Please could you add a test unit for it? I believe a feature test for this where you dispatch an action and check the searches state variable would work.